### ISSUE TITLE
Reduce load of iBeacon sensors

### ISSUE THREAD

**Issue by Ernst79** [custom-components/ble_monitor#672]
iBeacon sensors seem to be reporting quite frequently. We have only one iBeacon device, and it seems to report about 100 times per minute. On the Home Assistant forum, someone already had to turn off iBeacon devices, as HA got overloaded. 

```
2022-01-29 20:07:11 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 102 BLE ADV messages processed for 7 measuring device(s)
```

I think it would be good to filter this a bit, such that at least the state isn't updated every second, like is being done for the device_tracker. I would propose to use the `scan_interval` option, to only update the state of all the iBeacon (and AltBeacon) sensors once in this period. 

![image](https://user-images.githubusercontent.com/24375847/151674966-c7ded425-1b81-447e-901a-6256637b1ad7.png)
 
For the device tracker it works as follows: when it receives a BLE message, the next 20 seconds (= default scan_interval), it won't update the state.

**Comment by dmamontov** [comment]
This should definitely be adjusted, I already know many who generally set it to 0 for a more accurate definition. To be honest, it's not very clear what is loaded. I think it's the recorder with the default value.

**Comment by Ernst79** [comment]
The problem is probably with the uuid and mac sensors, which are now defined as `InstantUpdateSensor`, which means that the state is updated on every message. However, these sensor would hardly change, if at all. So, I don’t think there is a need to update these every message, once per period is probably ok. 

We could change it to the `MeasuringSensor` class, but we might have to define a separate class, that is just picking the last received state. The MeasuringSensor class is doing some averaging, which we don’t need for MAC and UUID. 

@dmamontov If you need help with this, let me know

**Comment by dmamontov** [comment]
uuid will most likely not change, but mac is constantly changing. Yes, I will.

**Comment by Ernst79** [comment]
Yes, but will it change more than acne per minute? Instantupdate sensor is now updating the state 100 times per minute, let’s reduce that to once per period.

anyway, thanks

**Comment by dmamontov** [comment]
I completely agree.

**Comment by Ernst79** [comment]
@dmamontov I'm testing your PR right now, it seems to work, I'll wait a little while till the mac changes, just to see that it works. 

However, I notice that the other sensors are also updating 100x per minute, like the temperature sensor. I'm not sure why, as you have used the MeasuringSensorClass. I have to dig into this if it are my settings, but could you check this for me on your end. Is the state of e.g. the temperature sensor updated every second, or just once per minute, doing the averaging.

**Comment by dmamontov** [comment]
You are right, the rest, in principle, work just as much per second. I propose to add a check for a change in state to all sensors.

**Comment by Ernst79** [comment]
Yes, that is ok, but the strange thing is that the temperature and humidity sensor normally averaging the results over one period (one minute), which might be better to filter out extreme values (sometimes happens on Xiaomi sensors e.g. It might be related to the fact that it is not defined by a mac, but by the UUID of the sensor, but that is just a guess.

**Comment by Ernst79** [comment]
Ah, I found it, they are in instant sensors. Wait a second please.

**Comment by Ernst79** [comment]
We just need to change this

```
    'iBeacon'                 : [["rssi", "measured power"], ["uuid", "mac", "major", "minor", "cypress temperature", "cypress humidity"], []], # mac can be dynamic
    'AltBeacon'               : [["rssi", "measured power"], ["uuid", "mac", "major", "minor"], []], # mac can be dynamic
```

to 

```
    'iBeacon'                 : [["rssi", "measured power", "cypress temperature", "cypress humidity"], ["uuid", "mac", "major", "minor"], []], # mac can be dynamic
    'AltBeacon'               : [["rssi", "measured power"], ["uuid", "mac", "major", "minor"], []], # mac can be dynamic
```

Not sure about major and minor, maybe better to use the StateChanged class there

**Comment by dmamontov** [comment]
Yes, I think that would be better. Added to PR.

**Comment by Ernst79** [comment]
Thanks, these iBeacon sensors seem to be sending a lot of data

![image](https://user-images.githubusercontent.com/24375847/151703815-ee8fac0d-58f7-4b3d-84e6-82d5c8018c13.png)

300 BLE advertisements every minute, that's 5 per second!!!

**Comment by dmamontov** [comment]
Here they are) it’s good that we are now deciding, because they asked me to add new species in the Russian-speaking community. But first, you need to redo the configuration a little, and make the code cleaner.

**Comment by Ernst79** [comment]
Sure, but I'm not a programmer, just an enthusiast, so any help or proposal is more than welcome. If you want, I can add you ass code owner, such that you can work on the repo directly.

**Comment by dmamontov** [comment]
I think it will be possible) Let's after the edits that I will send next week.

I have a lot of plans for the component, because on the data that he receives, I write a large component for triangulation with direction angle prediction) But this is still for a long time)

**Comment by Ernst79** [comment]
Ok, sounds great!. I've released it as 7.4.0, so we can close this one.
